source/Geometry/src/GeometricModule.cc

```diff
@@ -1,172 +1,210 @@
+//
+// Constructor base class
+//
+GeometricModule::GeometricModule() :
+    dsDistance("dsDistance", parsedAndChecked(), 0.),
+    physicalLength("physicalLength", parsedOnly(), 0.)
+{}
 
-// Distance between origin and module if hits
+// ??? method -> Distance between origin and module if hits
 // Otherwise -1
-double GeometricModule::trackCross(const XYZVector& PL, // Base line point
-                                   const XYZVector& PU) // Line direction
-{
-  double distance;
-
-  distance = triangleCross(basePoly().getVertex(0),
-                           basePoly().getVertex(1),
-                           basePoly().getVertex(2),
-                           PL, PU);
-
-  if (distance<0) {
-    distance = triangleCross(basePoly().getVertex(0),
-                             basePoly().getVertex(2),
-                             basePoly().getVertex(3),
-                             PL, PU);
-  }
-
-  if (distance>=0) {
-    numHits_++;
-    distance /= PU.r();
-  } else {
-    return -1;
-  }
-
-  return distance;
-
-}
-
-
-
-double GeometricModule::triangleCross(const XYZVector& P1, // Triangle points
-                                      const XYZVector& P2,
-                                      const XYZVector& P3,
-                                      const XYZVector& PL, // Base line point
-                                      const XYZVector& PU) // Line direction
-{
-  bool moduleHit = false;
-
-  // Triangle coordinates
-  // t - P1 = alpha * (P2-P1) + beta * (P3-P1)
-
-  // Line coordinates:
-  // r = PL + gamma * PU
-
-
-  // How to solve the generic problem:
-
-  // d = PL - P1
-  // A = (P2-P1, P3-P1, -PU)
-  // v = intersection point
-  TVectorD d = TVectorD(3);
-  TMatrixD A = TMatrixD(3, 3);
-  TVectorD v = TVectorD(3);
-
-  d(0)=PL.x()-P1.x();
-  d(1)=PL.y()-P1.y();
-  d(2)=PL.z()-P1.z();
-
-  A(0, 0)=P2.x()-P1.x();
-  A(0, 1)=P3.x()-P1.x();
-  A(0, 2)=-1*PU.x();
-  A(1, 0)=P2.y()-P1.y();
-  A(1, 1)=P3.y()-P1.y();
-  A(1, 2)=-1*PU.y();
-  A(2, 0)=P2.z()-P1.z();
-  A(2, 1)=P3.z()-P1.z();
-  A(2, 2)=-1*PU.z();
-
-  Double_t determ;
-  A.InvertFast(&determ);
-
-  // The matrix is invertible
-  if (determ!=0) {
-    // v = A^{-1} * d
-    v = A * d;
-    // v(0) = alpha : triangle local coordinate
-    // v(1) = beta  : triangle local coordinate
-    // v(2) = gamma : line coordinate
-
-    //     std::cout << "Alpha: " << v(0) << std::endl;
-    //     std::cout << "Beta:  " << v(1) << std::endl;
-    //     std::cout << "Gamma: " << v(2) << std::endl;
-
-    if (
-      (v(0)>=0)&&
-      (v(1)>=0)&&
-      ((v(0)+v(1))<=1)
-      ) {
-      moduleHit = true;
-    } else {
-      moduleHit = false;
-    }
-
-  } else {
-    // It does not cross the triangle
-    moduleHit=false;
-    std::cout << "Matrix is not invertible" << std::endl; // debug
-  }
-
-
-  if (moduleHit) {
-    return v(2);
-  } else {
-    return -1.;
-  }
-
-}
-
-
-
-
-
-
+//double GeometricModule::trackCross(const XYZVector& PL, // Base line point
```

Comment:
This code was originally there and no code of the tklayout uses that, hence I commented that out to avoid any clashes in the future. I didn't want to put the code directly away ... If you want, I can leave uncommented, but somebody in the future may start thinking, why it's there...